code/modules/surgery/organs/lungs.dm

```diff
@@ -102,28 +102,16 @@
 			H.throw_alert("not_enough_nitro", /atom/movable/screen/alert/not_enough_nitro)
 		return FALSE
 
+	for(var/gas_id in GLOB.meta_gas_info)
+		breath.assert_gas(gas_id)
+
+	if(H.wear_mask && H.wear_mask.clothing_flags & GAS_FILTERING && H.wear_mask.has_filter == TRUE)
+		breath = H.wear_mask.consume_filter(breath)
```

Comment:
Do you want the filters to work even when you've got a gas tank hooked up to you?

Reply:
i was thinking about that, but in the end i made it work with a tank too since the tanks could be contaminated if you take it from someone else